megalinter/utils.py

```diff
@@ -267,8 +269,8 @@ def file_contains(file_name: str, regex_object: Optional[Pattern[str]]) -> bool:
 
 def file_is_generated(file_name: str) -> bool:
     with open(file_name, "rb") as f:
-        content = f.read(1024)
-    return "@not-generated" not in content and b"@generated" in content
+        content = f.read(SIZE_MAX_SOURCEFILEHEADER)
+    return not ("@not-generated" in content) or b"@generated" in content
```

Comment:
Oh well, my original logic would have worked as well, since the first expression would be `False` in the optimizable case. I'll leave it at this.

Reply:
More specifically, the new condition, `not ("@not-generated" in content) or "@generated" in content`, incorrectly evaluates to `True` for any file that contains neither `"@not-generated"` nor `"@generated"` in the header. I agree that `a not in b` reads more clearly than `not (a in b)`. Evidently, [Sourcery performs this refactoring automatically](https://docs.sourcery.ai/Reference/Python/Default-Rules/de-morgan/).

Reply:
@Kurt-von-Laven is the current code ok in your opinion ?
cc @sanmai-NL

Reply:
@Kurt-von-Laven